mongodb_store_msgs/srv/MongoQueryMsg.srv

```diff
@@ -17,6 +17,8 @@ StringPairList message_query
 StringPairList meta_query
 # key/values to be turned into a sort query
 StringPairList sort_query
+# key/values to be turned into a projection query
+StringPairList projection_query
```

Comment:
I'm no expert in mongodb (maybe @hawesie is) but to me it seems that you change som interface definition here which will break compatibility with existing code? In that case, I'd be reluctant to merge this. Maybe, in this case, better offer a new service with a new signature instead of modifying an existing one?